In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns  

%matplotlib inline

In [2]:
smile = pd.read_csv("../data/SmileOn11282020.csv", engine = 'python', skiprows = 2)

In [3]:
smile.head()

CallReportNum              ReportVersion  LinkedToCallReportNum  \
0       81381827  SMILE ON 60+ Registration                    NaN   
1       81379177  SMILE ON 60+ Registration                    NaN   
2       81377435      SMILE ON 60+ Clinical                    NaN   
3       81375215      SMILE ON 60+ Clinical                    NaN   
4       81373934      SMILE ON 60+ Clinical                    NaN   

  CallDateAndTimeStart CallDateAndTimeEnd  CallLength  CallerNum  \
0     11/25/2020 16:46   11/25/2020 16:48           2    3859874   
1     11/25/2020 15:40   11/25/2020 15:55          15    3859755   
2     11/25/2020 15:02   11/25/2020 15:02           0    2518680   
3     11/25/2020 14:17   11/25/2020 14:17           0    3231923   
4     11/25/2020 13:50   11/25/2020 13:58           8         -1   

   PhoneWorkerNum  WasRealCall  WasHangup  ...  \
0          136529          NaN        NaN  ...   
1           82341          NaN        NaN  ...   
2          119506          NaN        NaN  ...   
3          143703          NaN        NaN  ...   
4          135611          NaN        NaN  ...   

         SMILE ON 60+ Screening - Treatment Urgency  \
0                                               NaN   
1                                               NaN   
2                                               NaN   
3                                               NaN   
4  Urgent Care - within next week-pain or infection   

   SMILE ON 60+ Screening - Untreated Decay  \
0                                       NaN   
1                                       NaN   
2                                       NaN   
3                                       NaN   
4                                       Yes   

         SMILE ON 60+ Site Information - Enroller ID  \
0  Matthew Walker Comprehensive Health Center Inc...   
1                                                NaN   
2          Friends in Need Health Center - Kingsport   
3          Christ Community Health Services- Frayser   
4                      Neighborhood Health - Lebanon   

   SMILE ON 60+ Site Information - Enroller Location  \
0                                      Provider Site   
1                               Health Assist Office   
2                                      Provider Site   
3                                      Provider Site   
4                                                NaN   

   SMILE ON 60+ Treatment Plan - Action: Upload After Picture on Finish Tab  \
0                                                NaN                          
1                                                NaN                          
2                                                NaN                          
3                                                NaN                          
4                                                NaN                          

   SMILE ON 60+ Treatment Plan - Is the treatment plan completed?  \
0                                                NaN                
1                                                NaN                
2                                                Yes                
3                                                NaN                
4                                                NaN                

  SMILE ON 60+ Treatment Plan - Was a treatment plan developed?  \
0                                                NaN              
1                                                NaN              
2                                                Yes              
3                                                NaN              
4                                                Yes              

  SMILE ON 60+ Treatment Plan - Was the enrollee able to be restored to function (can chew) and "social six" esthetics (top front six teeth are present and disease free)  \
0                                                NaN                                                                  

In [4]:
smile.shape

(40032, 184)

In [5]:
 smile.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 40032 entries, 0 to 40031
Columns: 184 entries, CallReportNum to SMILE ON 60+ Verification of Eligibility - Verification of Eligibility
dtypes: float64(103), int64(6), object(75)
memory usage: 56.2+ MB


In [6]:
smile.tail()

CallReportNum              ReportVersion  LinkedToCallReportNum  \
40027       53192452  SMILE ON 60+ Registration                    NaN   
40028       67915561  SMILE ON 60+ Registration                    NaN   
40029       67251405      SMILE ON 60+ Clinical                    NaN   
40030       63063150      SMILE ON 60+ Clinical                    NaN   
40031       70839903      SMILE ON 60+ Clinical                    NaN   

      CallDateAndTimeStart CallDateAndTimeEnd  CallLength  CallerNum  \
40027      5/23/2018 14:11    5/23/2018 14:11           0         -1   
40028       5/15/2018 9:06     5/15/2018 9:10           4    3239813   
40029       5/10/2018 8:25     5/10/2018 8:26           1    2631417   
40030        5/1/2018 8:14      5/1/2018 8:24          10         -1   
40031       2/14/2018 8:10     2/14/2018 8:18           8         -1   

       PhoneWorkerNum  WasRealCall  WasHangup  ...  \
40027           60132          NaN        NaN  ...   
40028          128174          NaN        NaN  ...   
40029          131347          NaN        NaN  ...   
40030          116749          NaN        NaN  ...   
40031          116749          NaN        NaN  ...   

       SMILE ON 60+ Screening - Treatment Urgency  \
40027                                         NaN   
40028                                         NaN   
40029                                         NaN   
40030                                         NaN   
40031                                         NaN   

       SMILE ON 60+ Screening - Untreated Decay  \
40027                                       NaN   
40028                                       NaN   
40029                                       NaN   
40030                                       NaN   
40031                                       NaN   

       SMILE ON 60+ Site Information - Enroller ID  \
40027                                          NaN   
40028               Church Health Center - Memphis   
40029    Friends in Need Health Center - Kingsport   
40030         Interfaith Dental Clinic - Nashville   
40031         Interfaith Dental Clinic - Nashville   

       SMILE ON 60+ Site Information - Enroller Location  \
40027                                                NaN   
40028                                      Provider Site   
40029                                      Provider Site   
40030                                                NaN   
40031                                                NaN   

       SMILE ON 60+ Treatment Plan - Action: Upload After Picture on Finish Tab  \
40027                                                NaN                          
40028                                                NaN                          
40029                                      Not available                          
40030                                                NaN                          
40031                                                NaN                          

       SMILE ON 60+ Treatment Plan - Is the treatment plan completed?  \
40027                                                NaN                
40028                                                NaN                
40029                                                 No                
40030                                                NaN                
40031                                                NaN                

      SMILE ON 60+ Treatment Plan - Was a treatment plan developed?  \
40027                                                NaN              
40028                                                NaN              
40029                                                Yes              
40030                                                NaN              
40031                                                NaN              

      SMILE ON 60+ Treatment Plan - Was the enrollee able to be restored to function (can chew) and "social six" esthetics (top

## Removal Of Unused Columns 

In [7]:
smile_v1 = smile.drop(columns = ['WasRealCall', 'WasHangup', 'WasSexCall','WasWrongNumber', 'WasPrankCall','WasSilentCall','GeoCode','GeoAssignment','CensusDivision','CensusTrack','CensusReportingArea','211Region','PhoneType','ThirdPartyName','ThirdPartyOrganization','ThirdPartyPhoneNumber','ThirdPartyAddress','ThirdPartyCity','ThirdPartyCounty','ThirdPartyStateProvince','ThirdPartyPostalCode','VolunteerComments','CallersFeedback','TextField2','TextField3','TextField4','TextField5','TextField6','TextField7','TextField8','TextField9','TextField10','Supervisor','IPAddress','OrgNum','VolIdentifier','LinkedToCallReportNum'])

In [8]:
smile_v1.shape

(40032, 147)

In [9]:
smile_v1.head()

CallReportNum              ReportVersion CallDateAndTimeStart  \
0       81381827  SMILE ON 60+ Registration     11/25/2020 16:46   
1       81379177  SMILE ON 60+ Registration     11/25/2020 15:40   
2       81377435      SMILE ON 60+ Clinical     11/25/2020 15:02   
3       81375215      SMILE ON 60+ Clinical     11/25/2020 14:17   
4       81373934      SMILE ON 60+ Clinical     11/25/2020 13:50   

  CallDateAndTimeEnd  CallLength  CallerNum  PhoneWorkerNum     CityName  \
0   11/25/2020 16:48           2    3859874          136529  Springfield   
1   11/25/2020 15:55          15    3859755           82341    Nashville   
2   11/25/2020 15:02           0    2518680          119506  Rogersville   
3   11/25/2020 14:17           0    3231923          143703      Memphis   
4   11/25/2020 13:58           8         -1          135611          NaN   

  CountyName StateProvince  ...  \
0  Robertson            TN  ...   
1   Davidson            TN  ...   
2    Hawkins            TN  ...   
3     Shelby            TN  ...   
4        NaN            TN  ...   

         SMILE ON 60+ Screening - Treatment Urgency  \
0                                               NaN   
1                                               NaN   
2                                               NaN   
3                                               NaN   
4  Urgent Care - within next week-pain or infection   

   SMILE ON 60+ Screening - Untreated Decay  \
0                                       NaN   
1                                       NaN   
2                                       NaN   
3                                       NaN   
4                                       Yes   

         SMILE ON 60+ Site Information - Enroller ID  \
0  Matthew Walker Comprehensive Health Center Inc...   
1                                                NaN   
2          Friends in Need Health Center - Kingsport   
3          Christ Community Health Services- Frayser   
4                      Neighborhood Health - Lebanon   

  SMILE ON 60+ Site Information - Enroller Location  \
0                                     Provider Site   
1                              Health Assist Office   
2                                     Provider Site   
3                                     Provider Site   
4                                               NaN   

   SMILE ON 60+ Treatment Plan - Action: Upload After Picture on Finish Tab  \
0                                                NaN                          
1                                                NaN                          
2                                                NaN                          
3                                                NaN                          
4                                                NaN                          

  SMILE ON 60+ Treatment Plan - Is the treatment plan completed?  \
0                                                NaN               
1                                                NaN               
2                                                Yes               
3                                                NaN               
4                                                NaN               

  SMILE ON 60+ Treatment Plan - Was a treatment plan developed?  \
0                                                NaN              
1                                                NaN              
2                                                Yes              
3                                                NaN              
4                                                Yes              

  SMILE ON 60+ Treatment Plan - Was the enrollee able to be restored to function (can chew) and "social six" esthetics (top front six teeth are present and disease free)  \
0                                                NaN                                                                                                                        
1          

### Removal of all columns other than "Yellow" in Data Dictionary

In [10]:
smile_onlyyellow = smile[['CallReportNum','CallerNum','SMILE ON 60+ Screening - Last Screening Date','SMILE ON 60+ Oral Care Encounter - What care was provided to the enrollee? (Select all that apply)','SMILE ON 60+ Site Information - Enroller ID','SMILE ON 60+ Treatment Plan - Is the treatment plan completed?','SMILE ON 60+ Treatment Plan - Was a treatment plan developed?','SMILE ON 60+ Treatment Plan - Was the enrollee able to be restored to function (can chew) and "social six" esthetics (top front six teeth are present and disease free)']]

In [11]:
smile_onlyyellow

CallReportNum  CallerNum SMILE ON 60+ Screening - Last Screening Date  \
0           81381827    3859874                                          NaN   
1           81379177    3859755                                          NaN   
2           81377435    2518680                                          NaN   
3           81375215    3231923                                          NaN   
4           81373934         -1                                   11/25/2020   
...              ...        ...                                          ...   
40027       53192452         -1                                          NaN   
40028       67915561    3239813                                          NaN   
40029       67251405    2631417                                   11/29/2018   
40030       63063150         -1                                          NaN   
40031       70839903         -1                                          NaN   

      SMILE ON 60+ Oral Care Encounter - What care was provided to the enrollee? (Select all that apply)  \
0                                                    NaN                                                   
1                                                    NaN                                                   
2      5650 - Add tooth to existing partial denture - 30                                                   
3      No Code - Denture step Mandibular: may be used...                                                   
4                                                    NaN                                                   
...                                                  ...                                                   
40027                                                NaN                                                   
40028                                                NaN                                                   
40029                                                NaN                                                   
40030                              Case Management - $50                                                   
40031  D5284 - Removable unilateral partial denture/o...                                                   

             SMILE ON 60+ Site Information - Enroller ID  \
0      Matthew Walker Comprehensive Health Center Inc...   
1                                                    NaN   
2              Friends in Need Health Center - Kingsport   
3              Christ Community Health Services- Frayser   
4                          Neighborhood Health - Lebanon   
...                                                  ...   
40027                                                NaN   
40028                     Church Health Center - Memphis   
40029          Friends in Need Health Center - Kingsport   
40030               Interfaith Dental Clinic - Nashville   
40031               Interfaith Dental Clinic - Nashville   

      SMILE ON 60+ Treatment Plan - Is the treatment plan completed?  \
0                                                    NaN               
1                                                    NaN               
2                                                    Yes               
3                                                    NaN               
4                                                    NaN               
...                                                  ...               
40027                                                NaN               
40028                                                NaN               
40029                                                 No               
40030                                                NaN               
40031                                                NaN               

      SMILE ON 60+ Treatment Plan - Was a treatment plan developed?  \
0                                                    NaN              
1                    

### Renaming of columns so to remove issue with "ON" within column title. 

In [12]:
smile_onlyyellow = smile_onlyyellow.rename(columns = {'SMILE ON 60+ Screening - Last Screening Date':'screening_date', 'SMILE ON 60+ Oral Care Encounter - What care was provided to the enrollee? (Select all that apply)':'care_provided', 'SMILE ON 60+ Site Information - Enroller ID':'site', 'SMILE ON 60+ Treatment Plan - Is the treatment plan completed?':'treatment_plan_complete','SMILE ON 60+ Treatment Plan - Was a treatment plan developed?':'treatment_plan_developed','SMILE ON 60+ Treatment Plan - Was the enrollee able to be restored to function (can chew) and "social six" esthetics (top front six teeth are present and disease free)':'restored_to_function'})

In [13]:
smile_onlyyellow

CallReportNum  CallerNum screening_date  \
0           81381827    3859874            NaN   
1           81379177    3859755            NaN   
2           81377435    2518680            NaN   
3           81375215    3231923            NaN   
4           81373934         -1     11/25/2020   
...              ...        ...            ...   
40027       53192452         -1            NaN   
40028       67915561    3239813            NaN   
40029       67251405    2631417     11/29/2018   
40030       63063150         -1            NaN   
40031       70839903         -1            NaN   

                                           care_provided  \
0                                                    NaN   
1                                                    NaN   
2      5650 - Add tooth to existing partial denture - 30   
3      No Code - Denture step Mandibular: may be used...   
4                                                    NaN   
...                                                  ...   
40027                                                NaN   
40028                                                NaN   
40029                                                NaN   
40030                              Case Management - $50   
40031  D5284 - Removable unilateral partial denture/o...   

                                                    site  \
0      Matthew Walker Comprehensive Health Center Inc...   
1                                                    NaN   
2              Friends in Need Health Center - Kingsport   
3              Christ Community Health Services- Frayser   
4                          Neighborhood Health - Lebanon   
...                                                  ...   
40027                                                NaN   
40028                     Church Health Center - Memphis   
40029          Friends in Need Health Center - Kingsport   
40030               Interfaith Dental Clinic - Nashville   
40031               Interfaith Dental Clinic - Nashville   

      treatment_plan_complete treatment_plan_developed restored_to_function  
0                         NaN                      NaN                  NaN  
1                         NaN                      NaN                  NaN  
2                         Yes                      Yes                  Yes  
3                         NaN                      NaN                  NaN  
4                         NaN                      Yes                   No  
...                       ...                      ...                  ...  
40027                     NaN                      NaN                  NaN  
40028                     NaN                      NaN                  NaN  
40029                      No                      Yes                  Yes  
40030                     NaN                      NaN                  NaN  
40031                     NaN                      NaN                  NaN  

[40032 rows x 8 columns]

In [14]:
smile_onlyyellow.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 40032 entries, 0 to 40031
Data columns (total 8 columns):
 #   Column                    Non-Null Count  Dtype 
---  ------                    --------------  ----- 
 0   CallReportNum             40032 non-null  int64 
 1   CallerNum                 40032 non-null  int64 
 2   screening_date            14892 non-null  object
 3   care_provided             24849 non-null  object
 4   site                      32801 non-null  object
 5   treatment_plan_complete   17535 non-null  object
 6   treatment_plan_developed  17028 non-null  object
 7   restored_to_function      14276 non-null  object
dtypes: int64(2), object(6)
memory usage: 2.4+ MB


## Create CSV file to search for care provided code description

In [15]:
#smile_onlyyellow.care_provided.to_csv('../data/smile_onlyyellow_care_provided.csv')

In [16]:
smile_onlyyellow.treatment_plan_developed.value_counts(ascending= False)

Yes    15930
No      1098
Name: treatment_plan_developed, dtype: int64

In [17]:
smile_onlyyellow.treatment_plan_complete.value_counts()

No     12431
Yes     5104
Name: treatment_plan_complete, dtype: int64

In [18]:
smile_onlyyellow.CallerNum.value_counts()

-1          70
 2441776    36
 2431132    31
 2416076    29
 2466994    28
            ..
 3825101     1
 3743280     1
 2653820     1
 3686036     1
 3821558     1
Name: CallerNum, Length: 8708, dtype: int64

### Separating out most frequent CallerNum

In [19]:
smile_2441776 = smile_onlyyellow.loc[smile_onlyyellow['CallerNum'] == 2441776]

In [20]:
smile_2441776.head(10)

CallReportNum  CallerNum screening_date  \
685         80878764    2441776      9/25/2018   
1024        80784223    2441776      9/25/2018   
3353        79037448    2441776      9/25/2018   
3812        78220913    2441776       9/2/2020   
3833        78698465    2441776      9/25/2018   
5312        77631129    2441776      9/25/2018   
5906        76991587    2441776      9/25/2018   
6439        76991448    2441776      9/25/2018   
11473       71557178    2441776      9/25/2018   
14542       70184765    2441776      9/25/2018   

                                           care_provided  \
685    2331 - Resin two surfaces - Anterior - 15; 239...   
1024   No Code - Denture step Mandibular: may be used...   
3353   No Code - Denture step Mandibular: may be used...   
3812                                                 NaN   
3833   220 - Intraoral - Periapical First Film - 5; 1...   
5312   220 - Intraoral - Periapical First Film - 5; 3...   
5906   5110 - Complete denture-maxillary - 60; 5226 -...   
6439   5850 - Denture tissue conditioning maxillary - 15   
11473  1110 - Prophylaxis-adult - 45; 1330 - Oral Hyg...   
14542  No Code - Denture step Mandibular: may be used...   

                                       site treatment_plan_complete  \
685    Interfaith Dental Clinic - Nashville                      No   
1024   Interfaith Dental Clinic - Nashville                      No   
3353   Interfaith Dental Clinic - Nashville                      No   
3812   Interfaith Dental Clinic - Nashville                     NaN   
3833   Interfaith Dental Clinic - Nashville                      No   
5312   Interfaith Dental Clinic - Nashville                      No   
5906   Interfaith Dental Clinic - Nashville                      No   
6439   Interfaith Dental Clinic - Nashville                      No   
11473  Interfaith Dental Clinic - Nashville                     Yes   
14542  Interfaith Dental Clinic - Nashville                      No   

      treatment_plan_developed restored_to_function  
685                        Yes                   No  
1024                       Yes                   No  
3353                       Yes                   No  
3812                       Yes                  NaN  
3833                       Yes                   No  
5312                       Yes                  Yes  
5906                       Yes                  Yes  
6439                       Yes                  Yes  
11473                      Yes                  Yes  
14542                      Yes                  Yes

## Why does he have so many listings for the same day? And at such different CallReportNum? 

In [21]:
smile_2441776.CallReportNum.value_counts()


69171583    1
70184765    1
61865691    1
77631129    1
76991448    1
56542940    1
65850728    1
62797843    1
60706642    1
58800629    1
61582513    1
63277516    1
64563999    1
79037448    1
56587268    1
66789188    1
63277569    1
62848924    1
80784223    1
78698465    1
69739760    1
62797691    1
71557178    1
56543096    1
62172149    1
56542681    1
78220913    1
57786861    1
56596645    1
80878764    1
68129507    1
67757927    1
64343269    1
56570148    1
76991587    1
58800768    1
Name: CallReportNum, dtype: int64

In [22]:
smile_2431132 = smile_onlyyellow.loc[smile_onlyyellow['CallerNum'] == 2431132]

In [23]:
smile_2431132.head(10)

CallReportNum  CallerNum screening_date  \
766        80572613    2431132      11/4/2020   
1609       79831882    2431132            NaN   
2779       79039925    2431132            NaN   
3919       78182738    2431132            NaN   
5231       77249056    2431132            NaN   
5317       77203328    2431132            NaN   
6359       76456276    2431132            NaN   
7022       75903370    2431132            NaN   
7625       75329200    2431132            NaN   
8303       74795958    2431132            NaN   

                                          care_provided  \
766                                                 NaN   
1609  7140 - Extraction - erupted tooth or exposed r...   
2779  No Code - Denture step Mandibular: may be used...   
3919  No Code - Denture step Mandibular: may be used...   
5231  2330 - Resin one surface - Anterior - 15; 2331...   
5317                      1110 - Prophylaxis-adult - 45   
6359  140 - Limited Oral evaluation - problem focuse...   
7022  140 - Limited Oral evaluation - problem focuse...   
7625  7140 - Extraction - erupted tooth or exposed r...   
8303  7140 - Extraction - erupted tooth or exposed r...   

                                 site treatment_plan_complete  \
766   Karis Dental Clinic - Cleveland                      No   
1609  Karis Dental Clinic - Cleveland                      No   
2779  Karis Dental Clinic - Cleveland                      No   
3919  Karis Dental Clinic - Cleveland                      No   
5231  Karis Dental Clinic - Cleveland                      No   
5317  Karis Dental Clinic - Cleveland                      No   
6359  Karis Dental Clinic - Cleveland                      No   
7022  Karis Dental Clinic - Cleveland                      No   
7625  Karis Dental Clinic - Cleveland                      No   
8303  Karis Dental Clinic - Cleveland                      No   

     treatment_plan_developed restored_to_function  
766                        No                   No  
1609                      NaN                   No  
2779                      NaN                   No  
3919                      NaN                   No  
5231                      NaN                   No  
5317                      NaN                   No  
6359                      Yes                   No  
7022                       No                   No  
7625                      NaN                   No  
8303                      NaN                   No

In [24]:
smile_2416076 = smile_onlyyellow.loc[smile_onlyyellow['CallerNum'] == 2416076]

In [25]:
smile_2416076.head(10)

CallReportNum  CallerNum screening_date  \
1193        80792354    2416076     10/26/2020   
4030        78500974    2416076            NaN   
5497        77329572    2416076            NaN   
13125       70810955    2416076            NaN   
14930       70080607    2416076            NaN   
16025       69737230    2416076            NaN   
18407       68190928    2416076            NaN   
22967       66380005    2416076            NaN   
24167       65850976    2416076            NaN   
26184       65220398    2416076            NaN   

                                           care_provided  \
1193   220 - Intraoral - Periapical First Film - 5; 2...   
4030   7140 - Extraction - erupted tooth or exposed r...   
5497   1110 - Prophylaxis-adult - 45; 1330 - Oral Hyg...   
13125  1354 - Interim caries arresting medicament app...   
14930  272 - Bitewings - two films - 5; 1110 - Prophy...   
16025           2330 - Resin one surface - Anterior - 15   
18407  2330 - Resin one surface - Anterior - 15; 9951...   
22967  1110 - Prophylaxis-adult - 45; 1330 - Oral Hyg...   
24167  No Code - Denture step Maxillary: may be used ...   
26184  No Code - Denture step Maxillary: may be used ...   

                                          site treatment_plan_complete  \
1193      Interfaith Dental Clinic - Nashville                      No   
4030      Interfaith Dental Clinic - Nashville                      No   
5497   Interfaith Dental Clinic - Murfreesboro                      No   
13125     Interfaith Dental Clinic - Nashville                      No   
14930     Interfaith Dental Clinic - Nashville                      No   
16025     Interfaith Dental Clinic - Nashville                      No   
18407     Interfaith Dental Clinic - Nashville                     NaN   
22967     Interfaith Dental Clinic - Nashville                     NaN   
24167     Interfaith Dental Clinic - Nashville                     NaN   
26184     Interfaith Dental Clinic - Nashville                      No   

      treatment_plan_developed restored_to_function  
1193                       Yes                   No  
4030                       Yes                   No  
5497                       Yes                   No  
13125                      Yes                   No  
14930                      Yes                   No  
16025                      Yes                  Yes  
18407                      NaN                  NaN  
22967                      NaN                  NaN  
24167                      NaN                  NaN  
26184                      Yes                   No

## Where are the recipients of this care most located?

In [26]:
smile_location = smile[['CallerNum','CityName','CountyName','StateProvince','PostalCode','SMILE ON 60+ Site Information - Enroller ID','SMILE ON 60+ Screening - Last Screening Date']]

In [27]:
smile_location = smile_location.rename(columns = {'SMILE ON 60+ Site Information - Enroller ID': 'site','SMILE ON 60+ Screening - Last Screening Date':'date'})

In [28]:
smile_location

CallerNum     CityName CountyName StateProvince  PostalCode  \
0        3859874  Springfield  Robertson            TN     37172.0   
1        3859755    Nashville   Davidson            TN     37212.0   
2        2518680  Rogersville    Hawkins            TN     37857.0   
3        3231923      Memphis     Shelby            TN     38104.0   
4             -1          NaN        NaN            TN         NaN   
...          ...          ...        ...           ...         ...   
40027         -1          NaN        NaN           NaN         NaN   
40028    3239813      Memphis     Shelby            TN     38111.0   
40029    2631417    Kingsport   Sullivan            TN     37664.0   
40030         -1          NaN        NaN            TN         NaN   
40031         -1          NaN        NaN            TN         NaN   

                                                    site        date  
0      Matthew Walker Comprehensive Health Center Inc...         NaN  
1                                                    NaN         NaN  
2              Friends in Need Health Center - Kingsport         NaN  
3              Christ Community Health Services- Frayser         NaN  
4                          Neighborhood Health - Lebanon  11/25/2020  
...                                                  ...         ...  
40027                                                NaN         NaN  
40028                     Church Health Center - Memphis         NaN  
40029          Friends in Need Health Center - Kingsport  11/29/2018  
40030               Interfaith Dental Clinic - Nashville         NaN  
40031               Interfaith Dental Clinic - Nashville         NaN  

[40032 rows x 7 columns]

## What County?

In [29]:
smile_location.CountyName.value_counts()

Davidson      9299
Shelby        7398
Rutherford    3837
Sullivan      2209
Bradley       1789
              ... 
Fulton           1
McNairy          1
TN               1
Walker           1
Desoto           1
Name: CountyName, Length: 104, dtype: int64

## What City?

In [30]:
smile_location.CityName.value_counts()

Nashville        6456
Memphis          6304
Murfreesboro     2435
Cleveland        1718
Knoxville        1461
                 ... 
Snowville           1
Pleasant Hill       1
Ten Mile            1
Bells               1
Whitesburg          1
Name: CityName, Length: 387, dtype: int64

# Demographic information

In [31]:
smile_demo = smile[['CallerNum','SMILE ON 60+ Demographic Information - Gender','SMILE ON 60+ Demographic Information - Household Income Level','SMILE ON 60+ Demographic Information - How did you first hear about the SMILE ON 60+ program?','SMILE ON 60+ Demographic Information - Preferred Language/Language Needed','SMILE ON 60+ Demographic Information - Race/Ethnicity','SMILE ON 60+ Demographic Information - What other assistance programs are you on?','SMILE ON 60+ Demographic Information - When is the last time you visited your dentist?','SMILE ON 60+ Demographic Information - # of People in Household','SMILE ON 60+ Demographic Information - Other Language']]

In [32]:
smile_demo

CallerNum SMILE ON 60+ Demographic Information - Gender  \
0        3859874                                          Male   
1        3859755                                          Male   
2        2518680                                        Female   
3        3231923                                          Male   
4             -1                                          Male   
...          ...                                           ...   
40027         -1                                           NaN   
40028    3239813                                          Male   
40029    2631417                                        Female   
40030         -1                                           NaN   
40031         -1                                           NaN   

      SMILE ON 60+ Demographic Information - Household Income Level  \
0                                           $0 - $21 280              
1                                           $0 - $21 280              
2                                           $0 - $21 280              
3                                           $0 - $21 280              
4                                                    NaN              
...                                                  ...              
40027                                                NaN              
40028                                       $0 - $21 280              
40029                                       $0 - $21 280              
40030                                                NaN              
40031                                                NaN              

      SMILE ON 60+ Demographic Information - How did you first hear about the SMILE ON 60+ program?  \
0                                    Dental Clinic Staff                                              
1                                    Dental Clinic Staff                                              
2                                    Dental Clinic Staff                                              
3                                    Dental Clinic Staff                                              
4                                                    NaN                                              
...                                                  ...                                              
40027                                                NaN                                              
40028                                Dental Clinic Staff                                              
40029                                Dental Clinic Staff                                              
40030                                                NaN                                              
40031                                                NaN                                              

      SMILE ON 60+ Demographic Information - Preferred Language/Language Needed  \
0                                                English                          
1                                                English                          
2                                                    NaN                          
3                                                    NaN                          
4                                                    NaN                          
...                                                  ...                          
40027                                                NaN                          
40028                                            English                          
40029                                                NaN                          
40030                                                NaN                          
40031                                                NaN                          

      SMILE ON 60+ Demographic Information - Race/Ethnicity  \
0                                              Caucasian      
1

In [33]:
smile_demo = smile_demo.rename(columns= {'SMILE ON 60+ Demographic Information - Gender':'gender','SMILE ON 60+ Demographic Information - Household Income Level':'household_income','SMILE ON 60+ Demographic Information - How did you first hear about the SMILE ON 60+ program?':'how_heard','SMILE ON 60+ Demographic Information - Preferred Language/Language Needed':'preferred_language','SMILE ON 60+ Demographic Information - Race/Ethnicity':'race_ethnicity','SMILE ON 60+ Demographic Information - What other assistance programs are you on?':'additonal_assistance_programs','SMILE ON 60+ Demographic Information - When is the last time you visited your dentist?':'last_visit','SMILE ON 60+ Demographic Information - # of People in Household':'persons_in_household','SMILE ON 60+ Demographic Information - Other Language':'other_language'})

In [34]:
smile_demo

CallerNum  gender household_income            how_heard  \
0        3859874    Male     $0 - $21 280  Dental Clinic Staff   
1        3859755    Male     $0 - $21 280  Dental Clinic Staff   
2        2518680  Female     $0 - $21 280  Dental Clinic Staff   
3        3231923    Male     $0 - $21 280  Dental Clinic Staff   
4             -1    Male              NaN                  NaN   
...          ...     ...              ...                  ...   
40027         -1     NaN              NaN                  NaN   
40028    3239813    Male     $0 - $21 280  Dental Clinic Staff   
40029    2631417  Female     $0 - $21 280  Dental Clinic Staff   
40030         -1     NaN              NaN                  NaN   
40031         -1     NaN              NaN                  NaN   

      preferred_language    race_ethnicity additonal_assistance_programs  \
0                English         Caucasian                      SSI/SSDI   
1                English         Caucasian                           NaN   
2                    NaN         Caucasian      SNAP; TennCare; SSI/SSDI   
3                    NaN  African American                           NaN   
4                    NaN  African American                           NaN   
...                  ...               ...                           ...   
40027                NaN               NaN                           NaN   
40028            English         Caucasian                          SNAP   
40029                NaN         Caucasian                      SSI/SSDI   
40030                NaN               NaN                           NaN   
40031                NaN               NaN                           NaN   

                last_visit persons_in_household other_language  
0             3 to 5 years                    1            NaN  
1             3 to 5 years                    1            NaN  
2      Less than 12 months                    1            NaN  
3      Less than 12 months                    2            NaN  
4                      NaN                  NaN            NaN  
...                    ...                  ...            ...  
40027                  NaN                  NaN            NaN  
40028         1 to 2 years                    1            NaN  
40029  Less than 12 months                    1            NaN  
40030                  NaN                  NaN            NaN  
40031                  NaN                  NaN            NaN  

[40032 rows x 10 columns]

## Breakdown by Gender 

In [35]:
smile_demo_gender_counts = smile_demo.gender.value_counts()
smile_demo_gender_counts

Female         25018
Male           14580
Transgender       12
Other              1
Name: gender, dtype: int64

## Predominant ethnicities

In [36]:
smile_demo_eth_counts = smile_demo.race_ethnicity.value_counts()
smile_demo_eth_counts

Caucasian                            22730
African American                     10496
Hispanic                              1771
Unknown                                640
Asian                                  574
American Indian and Alaska Native      129
Multi-racial                           105
Pacific Islander                         9
Name: race_ethnicity, dtype: int64

## Where did they hear about Smile 60

In [37]:
smile_demo_how_counts = smile_demo.how_heard.value_counts()
smile_demo_how_counts

Dental Clinic Staff                      22805
Other                                     5725
This encounter                            2488
Wisdom Tooth Project Presentation         1988
SMILE ON CDHC                             1438
Senior Center                             1431
TV                                         983
Flyer                                      511
Newspaper                                  258
Dental Hotline                             145
SHIP Navigator                              83
Radio                                       48
Health Assist                               39
Tennessee Alliance for Legal Services        4
Habitat for Humanity Greater Memphis         2
Name: how_heard, dtype: int64

## Additional Assistance Program enrollees

In [38]:
smile_demo_aap_counts = smile_demo.additonal_assistance_programs.value_counts()
smile_demo_aap_counts

SSI/SSDI                               9204
SNAP                                   2447
TennCare                               1415
SNAP; SSI/SSDI                         1307
SNAP; TennCare; SSI/SSDI               1084
SNAP; TennCare                         1018
TennCare; SSI/SSDI                     1007
Section 8                               655
SNAP; Section 8                         409
Section 8; SSI/SSDI                     354
SNAP; TennCare; Section 8               308
SNAP; TennCare; Section 8; SSI/SSDI     287
TennCare; Section 8; SSI/SSDI           189
SNAP; Section 8; SSI/SSDI               184
TennCare; Section 8                     136
Name: additonal_assistance_programs, dtype: int64

In [39]:
smile_demo_language_counts = smile_demo.preferred_language.value_counts()
smile_demo_language_counts

English    9629
Spanish    1349
Other       417
Arabic      324
Name: preferred_language, dtype: int64

## Isolating only those cases that had a tooth extracted. 

In [40]:
smile_location_extract = smile[['CallerNum','CityName','CountyName','StateProvince','PostalCode','SMILE ON 60+ Site Information - Enroller ID','SMILE ON 60+ Screening - Last Screening Date','SMILE ON 60+ Oral Care Encounter - What care was provided to the enrollee? (Select all that apply)']]
smile_location_extract = smile_location_extract.rename(columns = {'SMILE ON 60+ Site Information - Enroller ID': 'site','SMILE ON 60+ Screening - Last Screening Date':'date','SMILE ON 60+ Oral Care Encounter - What care was provided to the enrollee? (Select all that apply)':'care_provided'})

In [41]:
smile_location_extract

CallerNum     CityName CountyName StateProvince  PostalCode  \
0        3859874  Springfield  Robertson            TN     37172.0   
1        3859755    Nashville   Davidson            TN     37212.0   
2        2518680  Rogersville    Hawkins            TN     37857.0   
3        3231923      Memphis     Shelby            TN     38104.0   
4             -1          NaN        NaN            TN         NaN   
...          ...          ...        ...           ...         ...   
40027         -1          NaN        NaN           NaN         NaN   
40028    3239813      Memphis     Shelby            TN     38111.0   
40029    2631417    Kingsport   Sullivan            TN     37664.0   
40030         -1          NaN        NaN            TN         NaN   
40031         -1          NaN        NaN            TN         NaN   

                                                    site        date  \
0      Matthew Walker Comprehensive Health Center Inc...         NaN   
1                                                    NaN         NaN   
2              Friends in Need Health Center - Kingsport         NaN   
3              Christ Community Health Services- Frayser         NaN   
4                          Neighborhood Health - Lebanon  11/25/2020   
...                                                  ...         ...   
40027                                                NaN         NaN   
40028                     Church Health Center - Memphis         NaN   
40029          Friends in Need Health Center - Kingsport  11/29/2018   
40030               Interfaith Dental Clinic - Nashville         NaN   
40031               Interfaith Dental Clinic - Nashville         NaN   

                                           care_provided  
0                                                    NaN  
1                                                    NaN  
2      5650 - Add tooth to existing partial denture - 30  
3      No Code - Denture step Mandibular: may be used...  
4                                                    NaN  
...                                                  ...  
40027                                                NaN  
40028                                                NaN  
40029                                                NaN  
40030                              Case Management - $50  
40031  D5284 - Removable unilateral partial denture/o...  

[40032 rows x 8 columns]

In [42]:
extraction = [7140,7210,7240,7250]
#smile_location_extract = smile_location_extract.care_provided.isin(extraction)
#smile_location_extract.care_provided.unique()
#array([extraction])

smile_location_extract = smile_location_extract[(smile_location_extract['care_provided'].str.contains('7140|7210|7240|7250', na = False))]
                       #(smile_location_extract['care_provided'].str.contains('7210',na = False) |
                       #(smile_location_extract['care_provided'].str.contains('7240',na = False) |
                       #(smile_location_extract['care_provided'].str.contains('7250',na = False)] 

In [43]:
smile_location_extract.head(10)

CallerNum   CityName CountyName StateProvince  PostalCode  \
9      3858402    Memphis     Shelby            TN     38128.0   
13     3172525  Cleveland    Bradley            TN     37312.0   
61     3620854  Cleveland    Bradley            TN         NaN   
111    3720256  Cleveland    Bradley            TN     37312.0   
115    3844630    Memphis     Shelby            TN     38128.0   
126    3761667    Memphis     Shelby            TN     38114.0   
148    3052444  Covington     Tipton            TN     38019.0   
153    3056273    Memphis     Shelby            TN     38127.0   
174    3829843    Memphis     Shelby            TN     38109.0   
177    3211727    Memphis     Shelby            TN     38118.0   

                                          site        date  \
9    Christ Community Health Services- Raleigh  11/24/2020   
13             Karis Dental Clinic - Cleveland    2/7/2020   
61             Karis Dental Clinic - Cleveland   8/10/2020   
111            Karis Dental Clinic - Cleveland   9/25/2020   
115  Christ Community Health Services- Raleigh  11/19/2020   
126  Christ Community Health Services- Frayser         NaN   
148  Christ Community Health Services- Raleigh   12/3/2019   
153  Christ Community Health Services- Frayser         NaN   
174  Christ Community Health Services- Frayser         NaN   
177  Christ Community Health Services- Frayser         NaN   

                                         care_provided  
9    140 - Limited Oral evaluation - problem focuse...  
13   No Code - Denture step Mandibular: may be used...  
61   7140 - Extraction - erupted tooth or exposed r...  
111  7140 - Extraction - erupted tooth or exposed r...  
115  150 - Comprehensive Oral Evaluation -new or es...  
126  7140 - Extraction - erupted tooth or exposed r...  
148  No Code - Denture step Mandibular: may be used...  
153  7140 - Extraction - erupted tooth or exposed r...  
174  7140 - Extraction - erupted tooth or exposed r...  
177  7140 - Extraction - erupted tooth or exposed r...

In [44]:
smile_location_extract.shape

(3878, 8)

In [45]:
smile_location_extract_city_count = smile_location_extract.CityName.value_counts()
smile_location_extract_city_count

Memphis           750
Nashville         542
Cleveland         268
Murfreesboro      201
Kingsport         136
                 ... 
Ripley              1
Hickory Valley      1
Spencer             1
Whiteville          1
Leoma               1
Name: CityName, Length: 228, dtype: int64

In [46]:
smile_location_extract_city_count_top5 = smile_location_extract_city_count.head()
smile_location_extract_city_count_top5

Memphis         750
Nashville       542
Cleveland       268
Murfreesboro    201
Kingsport       136
Name: CityName, dtype: int64

### Memphis had the most tooth extractions with 750. 

In [47]:
smile_location_extract_zip_count = smile_location_extract.PostalCode.value_counts()
smile_location_extract_zip_count_top5 = smile_location_extract_zip_count.head()
smile_location_extract_zip_count_top5

37311.0    106
37323.0     99
37013.0     90
37130.0     89
37207.0     80
Name: PostalCode, dtype: int64

### Zip Code 37311 has the most extractions with 106

In [48]:
smile_location_extract_site_count = smile_location_extract.site.value_counts()
smile_location_extract_site_count

Interfaith Dental Clinic - Nashville                                                    552
Interfaith Dental Clinic - Murfreesboro                                                 354
Matthew Walker Comprehensive Health Center Inc. - Nashville                             348
Church Health Center - Memphis                                                          344
Karis Dental Clinic - Cleveland                                                         290
Interfaith Health Clinic of Knoxville - Knoxville                                       216
Healing Hands Health Center - Bristol                                                   174
Friends in Need Health Center - Kingsport                                               163
Keystone Dental Care Inc. - Johnson City                                                152
Christ Community Health Services- Hickory Hill                                          136
Christ Community Health Services- Broad Avenue                                  

### Most extractions were performed at Interfaith Dental Clinic - Nashville with 552. 

In [49]:
smile_location_extract_date_count = smile_location_extract.date.value_counts()
smile_location_extract_date_count

6/13/2019     33
8/28/2019     26
9/17/2019     18
9/5/2019      17
12/3/2018     15
              ..
12/14/2019     1
12/15/2018     1
9/18/2020      1
12/16/2019     1
4/2/2020       1
Name: date, Length: 479, dtype: int64

### Most extractions were input on 13 June 2019

## Sorting for Top five (5) Sites that had function restored

In [50]:
smile_v1.head()

CallReportNum              ReportVersion CallDateAndTimeStart  \
0       81381827  SMILE ON 60+ Registration     11/25/2020 16:46   
1       81379177  SMILE ON 60+ Registration     11/25/2020 15:40   
2       81377435      SMILE ON 60+ Clinical     11/25/2020 15:02   
3       81375215      SMILE ON 60+ Clinical     11/25/2020 14:17   
4       81373934      SMILE ON 60+ Clinical     11/25/2020 13:50   

  CallDateAndTimeEnd  CallLength  CallerNum  PhoneWorkerNum     CityName  \
0   11/25/2020 16:48           2    3859874          136529  Springfield   
1   11/25/2020 15:55          15    3859755           82341    Nashville   
2   11/25/2020 15:02           0    2518680          119506  Rogersville   
3   11/25/2020 14:17           0    3231923          143703      Memphis   
4   11/25/2020 13:58           8         -1          135611          NaN   

  CountyName StateProvince  ...  \
0  Robertson            TN  ...   
1   Davidson            TN  ...   
2    Hawkins            TN  ...   
3     Shelby            TN  ...   
4        NaN            TN  ...   

         SMILE ON 60+ Screening - Treatment Urgency  \
0                                               NaN   
1                                               NaN   
2                                               NaN   
3                                               NaN   
4  Urgent Care - within next week-pain or infection   

   SMILE ON 60+ Screening - Untreated Decay  \
0                                       NaN   
1                                       NaN   
2                                       NaN   
3                                       NaN   
4                                       Yes   

         SMILE ON 60+ Site Information - Enroller ID  \
0  Matthew Walker Comprehensive Health Center Inc...   
1                                                NaN   
2          Friends in Need Health Center - Kingsport   
3          Christ Community Health Services- Frayser   
4                      Neighborhood Health - Lebanon   

  SMILE ON 60+ Site Information - Enroller Location  \
0                                     Provider Site   
1                              Health Assist Office   
2                                     Provider Site   
3                                     Provider Site   
4                                               NaN   

   SMILE ON 60+ Treatment Plan - Action: Upload After Picture on Finish Tab  \
0                                                NaN                          
1                                                NaN                          
2                                                NaN                          
3                                                NaN                          
4                                                NaN                          

  SMILE ON 60+ Treatment Plan - Is the treatment plan completed?  \
0                                                NaN               
1                                                NaN               
2                                                Yes               
3                                                NaN               
4                                                NaN               

  SMILE ON 60+ Treatment Plan - Was a treatment plan developed?  \
0                                                NaN              
1                                                NaN              
2                                                Yes              
3                                                NaN              
4                                                Yes              

  SMILE ON 60+ Treatment Plan - Was the enrollee able to be restored to function (can chew) and "social six" esthetics (top front six teeth are present and disease free)  \
0                                                NaN                                                                                                                        
1          

In [51]:
smile_location_func_resto = smile_v1.rename(columns = {'SMILE ON 60+ Screening - Last Screening Date':'screening_date', 'SMILE ON 60+ Oral Care Encounter - What care was provided to the enrollee? (Select all that apply)':'care_provided', 'SMILE ON 60+ Site Information - Enroller ID':'site', 'SMILE ON 60+ Treatment Plan - Is the treatment plan completed?':'treatment_plan_complete','SMILE ON 60+ Treatment Plan - Was a treatment plan developed?':'treatment_plan_developed','SMILE ON 60+ Treatment Plan - Was the enrollee able to be restored to function (can chew) and "social six" esthetics (top front six teeth are present and disease free)':'restored_to_function'})

In [52]:
smile_location_func_resto.head()

CallReportNum              ReportVersion CallDateAndTimeStart  \
0       81381827  SMILE ON 60+ Registration     11/25/2020 16:46   
1       81379177  SMILE ON 60+ Registration     11/25/2020 15:40   
2       81377435      SMILE ON 60+ Clinical     11/25/2020 15:02   
3       81375215      SMILE ON 60+ Clinical     11/25/2020 14:17   
4       81373934      SMILE ON 60+ Clinical     11/25/2020 13:50   

  CallDateAndTimeEnd  CallLength  CallerNum  PhoneWorkerNum     CityName  \
0   11/25/2020 16:48           2    3859874          136529  Springfield   
1   11/25/2020 15:55          15    3859755           82341    Nashville   
2   11/25/2020 15:02           0    2518680          119506  Rogersville   
3   11/25/2020 14:17           0    3231923          143703      Memphis   
4   11/25/2020 13:58           8         -1          135611          NaN   

  CountyName StateProvince  ...  \
0  Robertson            TN  ...   
1   Davidson            TN  ...   
2    Hawkins            TN  ...   
3     Shelby            TN  ...   
4        NaN            TN  ...   

         SMILE ON 60+ Screening - Treatment Urgency  \
0                                               NaN   
1                                               NaN   
2                                               NaN   
3                                               NaN   
4  Urgent Care - within next week-pain or infection   

   SMILE ON 60+ Screening - Untreated Decay  \
0                                       NaN   
1                                       NaN   
2                                       NaN   
3                                       NaN   
4                                       Yes   

                                                site  \
0  Matthew Walker Comprehensive Health Center Inc...   
1                                                NaN   
2          Friends in Need Health Center - Kingsport   
3          Christ Community Health Services- Frayser   
4                      Neighborhood Health - Lebanon   

  SMILE ON 60+ Site Information - Enroller Location  \
0                                     Provider Site   
1                              Health Assist Office   
2                                     Provider Site   
3                                     Provider Site   
4                                               NaN   

   SMILE ON 60+ Treatment Plan - Action: Upload After Picture on Finish Tab  \
0                                                NaN                          
1                                                NaN                          
2                                                NaN                          
3                                                NaN                          
4                                                NaN                          

  treatment_plan_complete treatment_plan_developed restored_to_function  \
0                     NaN                      NaN                  NaN   
1                     NaN                      NaN                  NaN   
2                     Yes                      Yes                  Yes   
3                     NaN                      NaN                  NaN   
4                     NaN                      Yes                   No   

   SMILE ON 60+ Verification of Eligibility - Transportation Arranged  \
0                                                NaN                    
1                                                NaN                    
2                                                NaN                    
3                                                NaN                    
4                                                NaN                    

  SMILE ON 60+ Verification of Eligibility - Verification of Eligibility  
0                                                NaN                      
1                                                NaN                      
2                                                NaN  

In [53]:
smile_location_func_resto.columns

Index(['CallReportNum', 'ReportVersion', 'CallDateAndTimeStart',
       'CallDateAndTimeEnd', 'CallLength', 'CallerNum', 'PhoneWorkerNum',
       'CityName', 'CountyName', 'StateProvince',
       ...
       'SMILE ON 60+ Screening - Treatment Urgency',
       'SMILE ON 60+ Screening - Untreated Decay', 'site',
       'SMILE ON 60+ Site Information - Enroller Location',
       'SMILE ON 60+ Treatment Plan - Action: Upload After Picture on Finish Tab',
       'treatment_plan_complete', 'treatment_plan_developed',
       'restored_to_function',
       'SMILE ON 60+ Verification of Eligibility - Transportation Arranged',
       'SMILE ON 60+ Verification of Eligibility - Verification of Eligibility'],
      dtype='object', length=147)

In [54]:
smile_location_func_resto = smile_location_func_resto[['CallerNum','CityName','CountyName','PostalCode','screening_date','site','treatment_plan_complete','treatment_plan_developed','restored_to_function']]

In [55]:
smile_location_func_resto.head()

CallerNum     CityName CountyName  PostalCode screening_date  \
0    3859874  Springfield  Robertson     37172.0            NaN   
1    3859755    Nashville   Davidson     37212.0            NaN   
2    2518680  Rogersville    Hawkins     37857.0            NaN   
3    3231923      Memphis     Shelby     38104.0            NaN   
4         -1          NaN        NaN         NaN     11/25/2020   

                                                site treatment_plan_complete  \
0  Matthew Walker Comprehensive Health Center Inc...                     NaN   
1                                                NaN                     NaN   
2          Friends in Need Health Center - Kingsport                     Yes   
3          Christ Community Health Services- Frayser                     NaN   
4                      Neighborhood Health - Lebanon                     NaN   

  treatment_plan_developed restored_to_function  
0                      NaN                  NaN  
1                      NaN                  NaN  
2                      Yes                  Yes  
3                      NaN                  NaN  
4                      Yes                   No

In [56]:
smile_location_func_resto.restored_to_function.value_counts()

No     8127
Yes    6149
Name: restored_to_function, dtype: int64

In [57]:
smile_location_func_resto_yes = smile_location_func_resto.loc[smile_location_func_resto['restored_to_function'] =="Yes"]

In [58]:
smile_location_func_resto_yes.head()

CallerNum     CityName CountyName  PostalCode screening_date  \
2     2518680  Rogersville    Hawkins     37857.0            NaN   
13    3172525    Cleveland    Bradley     37312.0       2/7/2020   
27    3856533     Bartlett     Shelby     38134.0     11/24/2020   
34    2835036      Memphis     Shelby     38127.0            NaN   
48    2928034      Memphis     Shelby     38118.0      10/1/2019   

                                         site treatment_plan_complete  \
2   Friends in Need Health Center - Kingsport                     Yes   
13            Karis Dental Clinic - Cleveland                      No   
27  Christ Community Health Services- Raleigh                      No   
34  Christ Community Health Services- Raleigh                     Yes   
48  Christ Community Health Services- Raleigh                     Yes   

   treatment_plan_developed restored_to_function  
2                       Yes                  Yes  
13                      NaN                  Yes  
27                      Yes                  Yes  
34                       No                  Yes  
48                      Yes                  Yes

In [59]:
smile_location_func_resto_yes.shape

(6149, 9)

In [60]:
smile_location_func_resto_yes_sort_date = smile_location_func_resto_yes.sort_values('screening_date').drop_duplicates(subset=['CallerNum'], keep='last')

In [61]:
smile_location_func_resto_yes_sort_date.head(10)

CallerNum         CityName CountyName  PostalCode screening_date  \
32169    2576785          Memphis     Shelby     38104.0      1/10/2019   
15001    3118094          Memphis     Shelby     38109.0      1/10/2020   
8334     3112474        Cleveland    Bradley     37323.0      1/10/2020   
15003    3118060          Cordova     Shelby     38016.0      1/10/2020   
21571    2652995          Antioch   Davidson         NaN      1/11/2019   
35064    2569210  South Pittsburg     Marion     37380.0      1/12/2019   
14922    3130588          Memphis     Shelby     38106.0      1/13/2020   
14883    3121806          Memphis     Shelby     38108.0      1/13/2020   
7815     3121996          Memphis     Shelby     38105.0      1/13/2020   
11201    3120760           Camden     Benton     38320.0      1/13/2020   

                                                    site  \
32169                     Church Health Center - Memphis   
15001     Christ Community Health Services- Broad Avenue   
8334                     Karis Dental Clinic - Cleveland   
15003     Christ Community Health Services- Broad Avenue   
21571             Neighborhood Health - Napier/Nashville   
35064  Smiles Inc. - South Pittsburg - serving Marion...   
14922          Christ Community Health Services- Frayser   
14883     Christ Community Health Services- Broad Avenue   
7815                      Church Health Center - Memphis   
11201                       Matthew Walker - Clarksville   

      treatment_plan_complete treatment_plan_developed restored_to_function  
32169                     Yes                      Yes                  Yes  
15001                      No                      Yes                  Yes  
8334                      Yes                      NaN                  Yes  
15003                      No                      Yes                  Yes  
21571                     Yes                      Yes                  Yes  
35064                     Yes                      Yes                  Yes  
14922                      No                      Yes                  Yes  
14883                      No                      Yes                  Yes  
7815                      Yes                      Yes                  Yes  
11201                     Yes                      Yes                  Yes

In [62]:
smile_location_func_resto_yes_sort_date.CallerNum.value_counts()

3501070    1
2956122    1
2728717    1
2659083    1
2759218    1
          ..
2733396    1
2886995    1
3845455    1
3056974    1
2893825    1
Name: CallerNum, Length: 3356, dtype: int64

In [63]:
smile_location_func_resto_vc = smile_location_func_resto_yes_sort_date.site.value_counts()
smile_location_func_resto_vc

Interfaith Dental Clinic - Nashville                                                    443
Church Health Center - Memphis                                                          433
Interfaith Dental Clinic - Murfreesboro                                                 299
Matthew Walker Comprehensive Health Center Inc. - Nashville                             275
Healing Hands Health Center - Bristol                                                   225
Interfaith Health Clinic of Knoxville - Knoxville                                       223
Friends in Need Health Center - Kingsport                                               196
Karis Dental Clinic - Cleveland                                                         194
Keystone Dental Care Inc. - Johnson City                                                180
Christ Community Health Services- Hickory Hill                                          103
Christ Community Health Services- Broad Avenue                                  

### There were 443 "Restored to Function" Cases at Interfaith Dental CLinic -Nashville

In [64]:
smile_location_func_resto_vc_top5 = smile_location_func_resto_vc.head()
smile_location_func_resto_vc_top5

Interfaith Dental Clinic - Nashville                           443
Church Health Center - Memphis                                 433
Interfaith Dental Clinic - Murfreesboro                        299
Matthew Walker Comprehensive Health Center Inc. - Nashville    275
Healing Hands Health Center - Bristol                          225
Name: site, dtype: int64

### Making listing Dataframe for the "No" Restored to Function

In [65]:
smile_location_func_resto_no = smile_location_func_resto.loc[smile_location_func_resto['restored_to_function'] =="No"]

In [66]:
smile_location_func_resto_no

CallerNum      CityName  CountyName  PostalCode screening_date  \
4             -1           NaN         NaN         NaN     11/25/2020   
35       3620854     Cleveland     Bradley         NaN      8/10/2020   
39       3707477     Cleveland     Bradley     37312.0      9/21/2020   
51       2879906       Memphis      Shelby     38108.0       9/6/2019   
53       3654343       Memphis      Shelby         NaN      8/24/2020   
...          ...           ...         ...         ...            ...   
39677    2399480  Murfreesboro  Rutherford     37130.0      8/14/2018   
39680    2399451  Murfreesboro  Rutherford     37129.0      8/15/2018   
39683    2399397     La Vergne  Rutherford     37086.0      8/14/2018   
39688    2399280  Murfreesboro  Rutherford     37130.0      8/14/2018   
39809    2390783       Madison    Davidson     37115.0       8/9/2018   

                                                 site treatment_plan_complete  \
4                       Neighborhood Health - Lebanon                     NaN   
35                    Karis Dental Clinic - Cleveland                      No   
39                    Karis Dental Clinic - Cleveland                      No   
51     Christ Community Health Services- Broad Avenue                      No   
53     Christ Community Health Services- Broad Avenue                      No   
...                                               ...                     ...   
39677                                             NaN                      No   
39680                                             NaN                      No   
39683         Interfaith Dental Clinic - Murfreesboro                      No   
39688         Interfaith Dental Clinic - Murfreesboro                      No   
39809         Interfaith Dental Clinic - Murfreesboro                     NaN   

      treatment_plan_developed restored_to_function  
4                          Yes                   No  
35                         NaN                   No  
39                         NaN                   No  
51                         Yes                   No  
53                         Yes                   No  
...                        ...                  ...  
39677                      Yes                   No  
39680                      Yes                   No  
39683                      Yes                   No  
39688                      Yes                   No  
39809                       No                   No  

[8127 rows x 9 columns]

In [67]:
smile_location_func_resto_no_sort_date = smile_location_func_resto_no.sort_values('screening_date').drop_duplicates(subset=['CallerNum'], keep='last')

In [68]:
smile_location_func_resto_no_sort_date.head(10)

CallerNum     CityName  CountyName  PostalCode screening_date  \
32923    2578404  Greeneville      Greene     37743.0       1/1/2019   
26337    2581726      Antioch    Davidson     37013.0      1/10/2019   
33623    2569064    Kingsport    Sullivan     37664.0      1/10/2019   
23863    2576785      Memphis      Shelby     38104.0      1/10/2019   
3465     3118060      Cordova      Shelby     38016.0      1/10/2020   
14996    3126326      Memphis      Shelby     38141.0      1/10/2020   
14185    3112474    Cleveland     Bradley     37323.0      1/10/2020   
35067    2599600  Nolensville  Williamson     37135.0      1/11/2019   
35088    2572761    Hermitage    Davidson     37076.0      1/11/2019   
33064    2554846       Smyrna  Rutherford     37167.0      1/11/2019   

                                                    site  \
32923           Keystone Dental Care Inc. - Johnson City   
26337  Matthew Walker Comprehensive Health Center Inc...   
33623          Friends in Need Health Center - Kingsport   
23863                     Church Health Center - Memphis   
3465      Christ Community Health Services- Broad Avenue   
14996     Christ Community Health Services- Hickory Hill   
14185                    Karis Dental Clinic - Cleveland   
35067  Matthew Walker Comprehensive Health Center Inc...   
35088  Matthew Walker Comprehensive Health Center Inc...   
33064                       Matthew Walker - Clarksville   

      treatment_plan_complete treatment_plan_developed restored_to_function  
32923                      No                      Yes                   No  
26337                      No                      Yes                   No  
33623                      No                      Yes                   No  
23863                      No                      Yes                   No  
3465                       No                      Yes                   No  
14996                      No                      Yes                   No  
14185                      No                      NaN                   No  
35067                      No                      Yes                   No  
35088                      No                      Yes                   No  
33064                      No                      Yes                   No

In [69]:
smile_location_func_resto_no_sort_date.CallerNum.value_counts()

3501070    1
2932099    1
2897254    1
3081570    1
2837855    1
          ..
3733216    1
3518074    1
3308184    1
2835095    1
2625540    1
Name: CallerNum, Length: 3095, dtype: int64

In [70]:
smile_location_func_resto_no_vc = smile_location_func_resto_no_sort_date.site.value_counts()
smile_location_func_resto_no_vc

Matthew Walker Comprehensive Health Center Inc. - Nashville                             404
Interfaith Dental Clinic - Nashville                                                    393
Church Health Center - Memphis                                                          300
Interfaith Dental Clinic - Murfreesboro                                                 279
Interfaith Health Clinic of Knoxville - Knoxville                                       231
Karis Dental Clinic - Cleveland                                                         202
Christ Community Health Services- Hickory Hill                                          149
Keystone Dental Care Inc. - Johnson City                                                117
Christ Community Health Services- Broad Avenue                                          117
Healing Hands Health Center - Bristol                                                   115
Friends in Need Health Center - Kingsport                                       

In [71]:
smile_location_func_resto_no_vc_top5 = smile_location_func_resto_no_vc.head()
smile_location_func_resto_no_vc_top5

Matthew Walker Comprehensive Health Center Inc. - Nashville    404
Interfaith Dental Clinic - Nashville                           393
Church Health Center - Memphis                                 300
Interfaith Dental Clinic - Murfreesboro                        279
Interfaith Health Clinic of Knoxville - Knoxville              231
Name: site, dtype: int64